bpf/lib/trace.h

```diff
@@ -246,13 +248,22 @@ _send_trace_notify(struct __ctx_buff *ctx, enum trace_point obs_point,
 			return;
 	}
 
+#ifdef ENABLE_WIREGUARD
+	if (THIS_INTERFACE_IFINDEX == WG_IFINDEX) {
```

Comment:
I'm not sure how llvm compiles this, but it's better to make sure this can be handled by preprocessor:
```diff
- #ifdef ENABLE_WIREGUARD
-	if (THIS_INTERFACE_IFINDEX == WG_IFINDEX) {
+ #if defined(ENABLE_WIREGUARD) && (THIS_INTERFACE_IFINDEX == WG_IFINDEX)
```

Reply:
`THIS_INTERFACE_IFINDEX` is a config variable, thus should not be evaluated by the preprocessor itself.
Unfortunately, if https://github.com/cilium/cilium/pull/38077 would land first we would not need this checks anymore.